xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteAction.java

```diff
@@ -240,7 +246,14 @@ private boolean deleteDocument(XWikiContext context, boolean shouldSkipRecycleBi
             doesAffectChildren(request, doc.getDocumentReference()) ? doc.getDocumentReference().getLastSpaceReference()
                 : doc.getTranslatedDocument(context).getDocumentReferenceWithLocale();
 
-        return deleteDocument(documentReference, context, shouldSkipRecycleBin);
+        return deleteDocument(documentReference, context, shouldSkipRecycleBin, newTarget, updateLinks, autoRedirect);
+    }
+
+    protected boolean deleteDocument(EntityReference entityReference, XWikiContext context,
```

Comment:
This makes it a new SPI. Is that absolutely needed?

Reply:
BTW requires a `@since` and I'd say `@Unstable` too

Reply:
I added this for https://github.com/xwiki/xwiki-platform/blob/master/xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/web/DeleteSpaceAction.java#L44
But I think I could drop this and just update directly the method call from `DeleteSpaceAction` to use the default values for the extra parameters.

Reply:
Can this method be considered as deprecated now? If this is the case, it can be moved to legacy (see https://dev.xwiki.org/xwiki/bin/view/Community/DevelopmentPractices#HDeprecatingCode)

Reply:
Yes, I think this should be done. I wasn't aware of the process. Thanks!